[docs] Add brownfield isolated approach guide (#43022)
expo/expo: docs/pages/brownfield/isolated-approach.mdx

For an alternative approach where React Native is integrated directly into your native project, see the [integrated approach guide](/brownfield/integrated-approach/).

## Prerequisites

To integrate React Native into your existing application, you'll need to set up a JavaScript development environment. This includes installing Node.js to run Expo CLI and Yarn to manage the project's JavaScript dependencies.

- [Node.js (LTS)](https://nodejs.org/en/): The runtime to execute JavaScript code and Expo CLI.
- [Yarn](https://yarnpkg.com/): A package manager for installing and managing JavaScript dependencies.

Learn more from the [Set up environment guide](/get-started/set-up-your-environment/).

## Setup an Expo project

<Step label="1">

### Create a new Expo project

<Terminal cmd={['$ npx create-expo-app@latest my-project']} />

This command creates a new directory named my-project that contains your new Expo project. While you can name the project anything, this guide uses my-project for consistency.

> This project does not need to live inside your existing native app and can be created in a separate repository or a monorepo. The new project includes an example TypeScript application to help you get started.

Reviewer: ```suggestion

Run the following command to create a new directory named **my-project** that contains your new Expo project. While you can name the project anything, this guide uses **my-project** for consistency.

<Terminal cmd={['$ npx create-expo-app@latest my-project']} />

The **my-project** does not need to live inside your existing native app and can be created in a separate repository or a monorepo. The new project includes an example TypeScript application to help you get started.
```

Let's have info on why need to we need to run the command first and then show the command.

</Step>

<Step label="2">
### Install expo-brownfield

Navigate to your new Expo project and install the `expo-brownfield` package, which provides the tools to build your React Native code as native libraries and integrate them into your existing native app.

<Terminal cmd={['$ npx expo install expo-brownfield']} />

</Step>

<Step label="3">
### Adjust the config plugin (optional)

`expo-brownfield` should automatically add an entry to the `plugins` array in your **app.json**, using the default configuration, which is sufficient for most projects.

```json app.json
{
  "expo": {
    "plugins": ["expo-brownfield"]
  }
}
```

The defaults are derived from your app config (for example, target names are based on your app's scheme or slug).
But you can also pass options to customize the target names, bundle identifiers, and publishing configuration.

```json app.json
{
  "expo": {
    "plugins": [
      [
        "expo-brownfield",
        {
          "ios": {
            "targetName": "MyBrownfield",
            "bundleIdentifier": "com.example.mybrownfield"
          },
          "android": {
            "libraryName": "mybrownfield",
            "group": "com.example",

>             "package": "com.example.mybrownfield",

Author: Let's keep without the `:1.0.0`

            "version": "1.0.0"
          }
        }
      ]
    ]
  }
}
```

Check the [API reference](/versions/latest/sdk/brownfield/) for details on all available options.

</Step>

## Export your Expo project as a native library

> Once you have your Expo project set up, use the `expo-brownfield` CLI to build your React Native code as XCFrameworks for iOS and AARs for Android.

Reviewer: ```suggestion
Once you have your Expo project set up, use the `expo-brownfield` CLI to build your React Native code as  AARs for Android and XCFrameworks for iOS.
```

<Tabs>
<Tab label="Android">

From your Expo project directory, run:

<Terminal cmd={['$ npx expo-brownfield build:android']} />

This will build the AAR and publish it to a Maven repository. By default, it publishes to your local Maven repository (`~/.m2`), but it can also be configured to publish to a remote repository.
The produced artifact name will be determined by your config plugin settings, in this case `com.username.myproject:brownfield:1.0.0`.

Check the [API reference](/versions/latest/sdk/brownfield/) for more details on build options, such as building only debug or release, specifying a custom output directory, and more.

</Tab>
<Tab label="iOS">

From your Expo project directory, run:

<Terminal cmd={['$ npx expo-brownfield build:ios']} />

This will build the XCFramework artifacts, compiling the framework target for both device and simulator architectures, package them into XCFrameworks, and copy the Hermes engine framework.

When the build process is completed, the output is placed in the **./artifacts** directory and contains:

- **\{TargetName\}.xcframework** - Your Expo project as a native library
- **hermesvm.xcframework** - The Hermes JavaScript engine

> Check the [API reference](/versions/latest/sdk/brownfield/) for more details on build options, such as building only debug or release, specifying a custom output directory, and more.

Reviewer: ```suggestion
See the [`expo-brownfield` API reference](/versions/latest/sdk/brownfield/) for more details on build options, such as building only debug or release, specifying a custom output directory, and more.
```

</Tab>
</Tabs>

<Collapsible summary="Debugging native targets">

If you need to debug native code of the Expo project targets, you can run `npx expo prebuild` to generate the native projects with the brownfield library targets, inside the `./android` and `./ios` directories.

<Terminal cmd={['$ npx expo prebuild']} />

This generates:

- **iOS**: A separate Xcode framework target containing `ReactNativeHostManager`, `ReactNativeViewController`, `ReactNativeView` (SwiftUI), `BrownfieldMessaging`, and `ReactNativeDelegate`.
- **Android**: A separate library module containing `ReactNativeHostManager`, `BrownfieldActivity`, `ReactNativeFragment`, `ReactNativeViewFactory`, and `BrownfieldMessaging`.

</Collapsible >

## Integrating into your native app

With the artifacts built, you can now integrate them into your existing native app. The exact steps will depend on your project structure and build system, but the general process involves adding the pre-built artifacts as dependencies and initializing the React Native host.

<Tabs>
<Tab label="Android">

<Step label="1">

> ### Add the Maven dependency

Reviewer: Right now, these headings inside the tab are part of the TOC in the right sidebar. I think it is confusing to see which heading is for Android or iOS tabs.

<img width="3142" height="838" alt="Image" src="https://github.com/user-attachments/assets/4b123409-8121-4b3d-b47e-176609a04f5e" />

I think we can use H4s here to ensure they don't become part of the TOC. Or use bold formatted text instead of headings.

Create an activity that extends `BrownfieldActivity` and use the `showReactNativeFragment()` extension:

```kotlin ExpoActivity.kt
import android.os.Bundle
import com.example.brownfield.BrownfieldActivity
import com.example.brownfield.showReactNativeFragment

class ExpoActivity : BrownfieldActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    showReactNativeFragment()
  }
}
```

Add the activity to your **AndroidManifest.xml** with a non-ActionBar theme:

```xml AndroidManifest.xml
<activity
  android:name=".ExpoActivity"
  android:theme="@style/Theme.AppCompat.Light.NoActionBar"
  android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize|uiMode"
/>
```

Then launch it from anywhere in your app:

```kotlin
startActivity(Intent(this, ExpoActivity::class.java))
```

`BrownfieldActivity` extends `AppCompatActivity` and handles forwarding configuration changes to Expo modules. The `showReactNativeFragment()` extension also sets up native back button handling automatically.

</Step>

</Tab>
<Tab label="iOS">

<Step label="1">
### Add XCFrameworks to your project

> Drag both XCFramework files (\{TargetName\}**.xcframework** and **hermes.xcframework**) into your Xcode project navigator. In the dialog that appears:

Reviewer: ```suggestion
Drag both XCFramework files (\{TargetName\}**.xcframework** and **hermesvm.xcframework**) into your Xcode project navigator. In the dialog that appears:
```

Should it be `hermesvm` as per line 119-120?
